libs/pubsub/pubsub.go

```diff
@@ -400,41 +366,38 @@ func (state *state) remove(clientID string, q Query) {
 	}
 }
 
-func (state *state) removeAll(clientID string) {
-	queryMap, ok := state.clientToQueryMap[clientID]
-	if !ok {
-		return
-	}
-
-	for qStr := range queryMap {
-		ch := state.queryToChanMap[qStr][clientID]
-		close(ch)
-
-		// remove the client from query map.
-		// if query has no other clients subscribed, remove it.
-		delete(state.queryToChanMap[qStr], clientID)
-		if len(state.queryToChanMap[qStr]) == 0 {
-			delete(state.queryToChanMap, qStr)
+func (state *state) removeClient(clientID string, reason error) {
+	for qStr, clientSubscriptions := range state.subscriptions {
```

Comment:
Why did we get rid of the clientToQueryMap? Just for simplicity? Is it worth the cost of looping over all queries to remove a client?

Reply:
For simplicity, yes. Plus we'll most probably get rid of pubsub, so that's fine.